x-pack/plugins/apm/server/lib/transactions/charts/get_anomaly_data/index.ts

```diff
@@ -44,12 +47,21 @@ export async function getAnomalySeries({
     return;
   }
 
+  // don't fetch anomalies when no specific environment is selected
+  if (!environment || environment === ENVIRONMENT_ALL.value) {
```

Comment:
If `environment` is no longer optional this should be changed, right?

```suggestion
  if (environment === ENVIRONMENT_ALL.value) {
```

Reply:
This is to guard against empty string `''`, but you're right, it might be overly defensive.

Reply:
Unless we actually do set `environment` to an empty string somewhere I don't think we need this check.